core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -419,4 +419,34 @@ class SocketServerTest extends JUnitSuite {
     assertEquals(Map.empty, nonZeroMetricNamesAndValues)
   }
 
+  @Test
+  def testProcessorMetricsTags(): Unit = {
+    // KafkaMetric metrics tagged with listener
+    val kafkaMetricNames = metrics
+      .metrics
+      .keySet.asScala
+      .filter(m => !m.tags.isEmpty)
+
+    assertFalse(kafkaMetricNames.isEmpty)
+    val expectedListeners = Set("PLAINTEXT","TRACE")
+    kafkaMetricNames.foreach { kafkaMetricName => {
+        System.err.println(kafkaMetricName.tags)
```

Comment:
oops thanks for spotting. Shame syserr/syout are't caught by something akin to a findbugs for Scala